File: 0919-projection-area-of-3d-shapes/0919-projection-area-of-3d-shapes.py
class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        n = len(grid)
        xy = 0
        xz = 0
        yz = 0
        for i in range(n):
            mx = grid[i][0]
            for j in range(n):
                if grid[i][j] > 0:
                    xy += 1
                if grid[i][j] > mx:
                    mx = grid[i][j]
            xz += mx
        for col in zip(*grid):
            yz += max(col)
        return xy + xz + yz

File: 0919-projection-area-of-3d-shapes/0919-projection-area-of-3d-shapes.java
class Solution {
    public int projectionArea(int[][] grid) {
        int n = grid.length;
        int xy = 0, xz = 0, yz = 0;
        for(int i = 0; i < n; i++) {
            int max = grid[i][0];
            for(int j = 0; j < n; j++){
                if(grid[i][j] > 0){
                    xy++;
                }
                if(grid[i][j] > max){
                    max = grid[i][j];
                }
            }
            xz += max;
        }
        for(int i = 0; i < n; i++) {
            int max = grid[0][i];
            for(int j = 0; j < n; j++) {
                if(grid[j][i] > max) {
                    max = grid[j][i];
                }
            }
            yz += max;
        }
        return xy + xz + yz;
    }
}